src/firmware/host/types.rs

```diff
@@ -0,0 +1,424 @@
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+pub struct Version {
+    /// The major version number.
+    pub major: u8,
+
+    /// The minor version number.
+    pub minor: u8,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[repr(C)]
+/// Certificates which are accepted for [`CertTableEntry`](self::CertTableEntry)
+pub enum SnpCertType {
+    /// AMD Root Signing Key (ARK) certificate
+    ARK,
+
+    /// AMD SEV Signing Key (ASK) certificate
+    ASK,
+
+    /// Versioned Chip Endorsement Key (VCEK) certificate
+    VCEK,
+
+    /// Other (Specify GUID)
+    OTHER(uuid::Uuid),
+
+    /// Empty or closing entry for the CertTable
+    Empty,
+}
+
+impl ToString for SnpCertType {
+    fn to_string(&self) -> String {
+        match self {
+            SnpCertType::ARK => "c0b406a4-a803-4952-9743-3fb6014cd0ae".to_string(),
+            SnpCertType::ASK => "4ab7b379-bbac-4fe4-a02f-05aef327c782".to_string(),
+            SnpCertType::VCEK => "63da758d-e664-4564-adc5-f4b93be8accd".to_string(),
+            SnpCertType::Empty => "00000000-0000-0000-0000-000000000000".to_string(),
+            SnpCertType::OTHER(guid) => guid.to_string(),
+        }
+    }
+}
+
+impl TryFrom<SnpCertType> for uuid::Uuid {
+    type Error = uuid::Error;
+    fn try_from(value: SnpCertType) -> Result<Self, Self::Error> {
+        match value {
+            SnpCertType::ARK => uuid::Uuid::parse_str(&SnpCertType::ARK.to_string()),
+            SnpCertType::ASK => uuid::Uuid::parse_str(&SnpCertType::ASK.to_string()),
+            SnpCertType::VCEK => uuid::Uuid::parse_str(&SnpCertType::VCEK.to_string()),
+            SnpCertType::Empty => uuid::Uuid::parse_str(&SnpCertType::Empty.to_string()),
+            SnpCertType::OTHER(guid) => Ok(guid),
+        }
+    }
+}
+
+impl TryFrom<&uuid::Uuid> for SnpCertType {
+    type Error = uuid::Error;
+
+    fn try_from(value: &uuid::Uuid) -> Result<Self, Self::Error> {
+        Ok(match value.to_string().as_str() {
+            "c0b406a4-a803-4952-9743-3fb6014cd0ae" => SnpCertType::ARK,
+            "4ab7b379-bbac-4fe4-a02f-05aef327c782" => SnpCertType::ASK,
+            "63da758d-e664-4564-adc5-f4b93be8accd" => SnpCertType::VCEK,
+            "00000000-0000-0000-0000-000000000000" => SnpCertType::Empty,
+            _ => SnpCertType::OTHER(*value),
+        })
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[repr(C)]
+/// An entry with information regarding a specific certificate.
+pub struct CertTableEntry {
+    /// A Specificy certificate type.
+    pub cert_type: SnpCertType,
+
+    /// The raw data of the certificate.
+    pub data: Vec<u8>,
+}
+
+impl CertTableEntry {
+    /// Façade for retreiving the GUID for the Entry.
+    pub fn guid_string(&self) -> String {
+        self.cert_type.clone().to_string()
+    }
```

Comment:
the clone isn't required I think?

```suggestion
        self.cert_type.to_string()
    }
```